src/sources/hydra.rs

```diff
@@ -226,19 +261,10 @@ impl gasket::framework::Worker<Stage> for Worker {
                 debug!("Hydra message: {}", text);
                 match serde_json::from_str::<HydraMessage>(text) {
                     Ok(hydra_message) => {
-                        // TODO: search for the intersection point to ensure
-                        // we're on the same chain.
-                        match &self.intersect {
-                            Point::Specific(slot, _hash) if &hydra_message.seq <= slot => {
-                                debug!(
-                                    "Skipping message {} before or at intersection {}",
-                                    hydra_message.seq, slot
-                                );
-                                Ok(())
-                            }
-                            Point::Origin | Point::Specific(_, _) => {
+                        if self.intersect.should_process(hydra_message.pseudo_point()) {
```

Comment:
I'm not too sure about `should_process` mutating the intersect worker state _before_ we call `process_next`. We could split it out as an explicit final step.

Reply:
Went in a different direction in [41915ea](https://github.com/Anviking/oura/pull/9/commits/41915ea3234fb1fe2c527a9ce54eca0eb64f1ae2) with Arnauld: `process` and a `process_message` helper instead of `process_next` and `should_process`.